common/src/main/java/io/netty/util/concurrent/CompleteFuture.java

```diff
@@ -23,10 +23,14 @@
 /**
  * A skeletal {@link Future} implementation which represents a {@link Future} which has been completed already.
  */
-public abstract class CompleteFuture<V> extends AbstractFuture<V> {
+public abstract class CompleteFuture<V> implements Future<V> {
 
     private final EventExecutor executor;
 
+    // It is fine to not make this volatile as even if we override the value in there it does not matter as
+    // DefaultFutureCompletionStage has no state itself and is just a wrapper around this CompletableFuture instance.
+    private DefaultFutureCompletionStage<V> stage;
```

Comment:
Can't tell from the PR but can this be `FutureCompletionStage`?

Reply:
if you change the local var type in io.netty.util.concurrent.CompleteFuture#asStage too yes.

Reply:
Same as https://github.com/netty/netty/pull/9004#discussion_r272661486